core/arch/arm/plat-imx/imx_huk.c

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/**
+ * @copyright 2019 NXP
+ *
+ * @file    imx_huk.c
+ *
+ * @brief   i.MX Hardware Unique Key generation.\n
+ *          Call CAAM operation to generate the key derived
+ *          from the master key
+ */
+/* Standard includes */
+#include <string.h>
+
+/* Global includes */
+#include <kernel/tee_common_otp.h>
+#include <utee_defines.h>
+
+/* Platform includes */
+#include <imx.h>
+
+/* Crypto API Extension includes */
+#include <crypto_extension.h>
+
+/* Local includes */
+
+/**
+ * @brief   Return a HW unique key value.\n
+ *          On i.MX device, return a derivation of the Master Key
+ *          by calling the CAAM Blob master key verification
+ *          operation using a key modifier corresponding of the
+ *          first 16 bytes of the Die ID
+ *
+ * @param[out] hwhuk  HW Unique key
+ */
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwhuk)
+{
+	TEE_Result ret;
+
+	struct drvcrypt_buf cryptohuk = {0};
+
+	/* Initialize the HUK value */
+	memset(hwhuk->data, 0, sizeof(hwhuk->data));
+
+	cryptohuk.length = sizeof(hwhuk->data);
+	cryptohuk.data   = hwhuk->data;
+
+	ret = crypto_generate_huk(&cryptohuk);
+
+	/*
+	 * If there is an error during the Master key derivation, let the device
+	 * booting with a 0's key
+	 */
+	if (ret != TEE_SUCCESS) {
```

Comment:
Maybe only hide errors when some sort of testing/development flag is set? If feels odd to allow the device to boot when the security module is known to be in a bad state.

My understanding of the CAAM is that this will fail if High Assurance Boot (HAB) is not enabled on the device, which is likely during development.  We allow a zero key through when using `CFG_RPMB_TESTKEY` since that is our primary use case for `tee_otp_get_hw_unique_key()`, although the system PTA now also offers a key derivation function via `system_derive_ta_unique_key()`...

Unfortunately I don't recall having come across a good config flag for that in OP-TEE which also covers the non-RPMB path (`tee_fs_init_key_manager()`).